plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -217,6 +217,9 @@ func setScheduling(a admission.Attributes, pod *api.Pod, runtimeClass *nodev1.Ru
 }
 
 func validateOverhead(a admission.Attributes, pod *api.Pod, runtimeClass *nodev1.RuntimeClass) (err error) {
+	if len(pod.Spec.Overhead) == 0 {
+		return nil
+	}
```

Comment:
it's a bit inconsistent we trust `pod` but we do check carefully `runtimeClass`, but OTOH we did the same in the original code

Reply:
Thank you for your review. Should I move `len(pod.Spec.Overhead) == 0` before the call to the `validateOverhead` function, or check it within the `else` block on line 232? My original intention was that if `Pod.Spec.Overhead` is empty or undefined, there's no need to check `RuntimeClass`, because the `Pod` isn't using `RuntimeClass`.

Reply:
> it's a bit inconsistent we trust `pod` but we do check carefully `runtimeClass`, but OTOH we did the same in the original code


Thank you for your review. Should I move len(pod.Spec.Overhead) == 0 before the call to the validateOverhead function, or check it within the else block on line 232? My original intention was that if Pod.Spec.Overhead is empty or undefined, there's no need to check RuntimeClass, because the Pod isn't using RuntimeClass.

Reply:
@ffromani  Hello,  Could you please provide some advice?

If I move `len(pod.Spec.Overhead) == 0` before calling `validateOverhead`, the test for `validateOverhead` will fail because there is no check for `pod.Spec.Overhead` within `func validateOverhead`. An alternative is to place `len(pod.Spec.Overhead) == 0` in the `else` block of `if runtimeClass != nil && runtimeClass.Overhead != nil` within `func validateOverhead`. However, I want to avoid repeating the `len(pod.Spec.Overhead) == 0` check in both `func validateOverhead` and `func Validate`. Another option is to remove some of the `func validateOverhead` test code related to pod overhead.

Reply:
What I'd do here is actually do nothing more. The existing code already assumes extensively `pod != nil`, and this change doesn't make things worse in this regard. Improving further would require review of how the admission flow operates; it's likely the flow ensures earlier `pod != nil` (and `pod.Spec != nil`). I'm happy enough with the changes, let's see what other reviewers/the approvers think.

Reply:
ok, thank you.